### PR TITLE
IonQ: Handle 409 errors that are injected by Cloudflare

### PR DESCRIPTION
**Pull request by Cynocracy**
When these occur on GETs, they can be retried.

### CODE REVIEW

File: cirq-ionq/cirq_ionq/ionq_client.py [hunk 1]
```diff
@@ -24,13 +24,23 @@
...
+# We may want to condition on the body as well, to allow for some GET requests to return 409 in
+# the future.
+RETRIABLE_FOR_GETS = {requests.codes.conflict}
+# Retriable regardless of the source
+# Handle 52x responses from cloudflare.
+# See https://support.cloudflare.com/hc/en-us/articles/115003011431/
+RETRIABLE_STATUS_CODES = {
+    requests.codes.internal_server_error,
+    requests.codes.service_unavailable,
+    *[i for i in range(520, 530)],
```
Comment (dabacon) [inline cirq-ionq/cirq_ionq/ionq_client.py:38]: nit: slightly cleaner `*list(range(520, 530))`

**Review by dabacon — APPROVED** [review]
An excellent job not selling me on cloudfare.

**Comment by Cynocracy** [comment]
I do my best ;) These errors are hopefully going away, but we're pretty set on not returning 409s ourselves on GETs, so ty for the review!

**Comment by Cynocracy** [comment]
Revived this zombie PR :zombie: 
Merged with https://github.com/quantumlib/Cirq/pull/6006

**Review by gmauricio — APPROVED** [review]

**Comment by Cynocracy** [comment]
test appears to have flaked

**Comment by Cynocracy** [comment]
Are the typescript tests just busted? (also there's typescript now? :open_mouth: )

**Comment by vtomole** [comment]
@Cynocracy  I have a fix here: https://github.com/quantumlib/Cirq/pull/6014

**Comment by Cynocracy** [comment]
Thanks Victory! This was actually a separate PR, was just noting the test failures

**Comment by ColemanCollins** [comment]
Reopening and updating from master; per jon the flaky tests were causing this to not be ready to merge but the underlying failure around cloudflare fronting the IonQ API still needs to be addressed

File: cirq-ionq/cirq_ionq/ionq_client_test.py [hunk 2]
```diff
@@ -246,6 +246,25 @@ def test_ionq_client_create_job_timeout(mock_post):
...
+    response1 = mock.MagicMock()
+    response2 = mock.MagicMock()
+    mock_get.side_effect = [response1, response2]
+    response1.ok = False
+    response1.status_code = requests.codes.conflict
+    response1.request.method = "GET"
+    response2.ok = True
+    response2.json.return_value = {'foo': 'bar'}
+
+    client = ionq.ionq_client._IonQClient(remote_host='http://example.com', api_key='to_my_heart')
```
Comment (ColemanCollins) [inline cirq-ionq/cirq_ionq/ionq_client_test.py:260]: @dabacon @Cynocracy I still love the dummy API key we use for these tests

Comment (Cynocracy) [reply cirq-ionq/cirq_ionq/ionq_client_test.py:260]: lol, it's pretty great

**Pull request merged**